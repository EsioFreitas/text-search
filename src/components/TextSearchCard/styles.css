.search-card {
  padding-top: 1rem;
  background-color: white;
  border-radius: 0.5rem;
  box-shadow: 1px 1px 5px 2px #ccc;
}

.search-card h3 {
  padding: 1rem;
  padding-bottom: 1.5rem;
}

.search-card > div {
  display: flex;
  padding: 1rem;
  justify-content: space-between;
  align-items: center;
  background-color: #eee;
  border-radius: 0 0 0.5rem 0.5rem;
}

.search-status {
  display: flex;
  align-items: center;
}

.search-status div {
  background-color: teal;
  height: 0.7rem;
  width: 0.7rem;
  display: inline-block;
  border-radius: 50%;
  margin-right: .3rem;
}
