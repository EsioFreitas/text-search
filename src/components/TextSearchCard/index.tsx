import React from "react";
import "./styles.css";

const TextSearchCard: React.FC = () => {
  return (
    <div className="search-card">
      <h3>Titulo do card</h3>
      <div>
        <div className="search-status">
          <div /> <span>Status</span>
        </div>
        <button>Visualizar</button>
      </div>
    </div>
  );
};

export default TextSearchCard;
