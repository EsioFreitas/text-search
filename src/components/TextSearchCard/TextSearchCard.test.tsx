import React from "react";
import { render } from "@testing-library/react";
import TextSearchCard from "./";

describe("TextSearchCard tests", () => {
  it("should TextSearchCard component return", () => {
    const { baseElement } = render(
      <TextSearchCard
        searchText={{
          id: "",
          status: "",
          urls: [],
          title: "",
        }}
        toggle={() => {}}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it("should render TextSearchCard component", async () => {
    const { container } = render(
      <TextSearchCard
        searchText={{
          id: "",
          status: "",
          urls: [],
          title: "",
        }}
        toggle={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
