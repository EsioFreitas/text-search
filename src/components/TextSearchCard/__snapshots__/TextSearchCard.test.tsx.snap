// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSearchCard tests should render TextSearchCard component 1`] = `
<div>
  <div
    class="search-card"
  >
    <h3 />
    <h4 />
    <div>
      <div
        class="search-status"
      >
        <div
          style="background-color: rgb(204, 204, 204);"
        />
        <span
          style="color: rgb(204, 204, 204);"
        />
      </div>
      <button>
        View
      </button>
    </div>
  </div>
</div>
`;
